hearpreprocess/sampler.py

```diff
@@ -59,45 +61,36 @@
         "task_config": dcase2016_task2.generic_task_config,
         "audio_sample_size": 4,
         "necessary_keys": [],
+        "get_download_and_extract_tasks": pipeline.get_download_and_extract_tasks,
```

Comment:
We need this because sometimes the download and extract tasks depend on the type of task. For example some tasks even have multipart url sand they have very different download and extract tasks. Similarly with tfds task, they have a different download tasks. So essentially if we pass that function in this dict, that helps in doing the sampling.

Reply:
I changed the names around, so that it looks fine. However, we cannot have one name, as then one will be global and the other one is local to the function and that will create ambiguity